buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy

```diff
@@ -128,11 +136,37 @@ class DistributedPerformanceTest extends ReportGenerationPerformanceTest {
             e.printStackTrace()
             throw e
         } finally {
-            generatePerformanceReport()
+            writeBinaryResults()
+            if(isSuccessfulFirstRun() || isRerun()) {
+                generatePerformanceReport()
```

Comment:
We don't always have a re-runner, don't we need to create the report for the initial build as well if we don't rerun the build step even if the first step failed?